homeassistant/components/stream/hls.py

```diff
@@ -79,12 +79,12 @@ def render_playlist(track):
         """Render playlist."""
         segments = track.segments
 
-        if not segments:
+        if not segments or len(segments) < NUM_HLS_SEGMENTS:
```

Comment:
Does it still make sense to make a playlist even if, say, only 1 segment is available?
Below could be `segments[_NUM_HLS_SEGMENTS:]` to handle that case gracefully.

Reply:
I don't think it really makes sense to have playlists with 1 segment (I think all the players use a few segments for buffering), but the current code allows for it, and it might get the clients to start loading segments earlier, so we can stick to that as long as we don't see any problems.